pkg/controllers/user/monitoring/clusterHandler.go

```diff
@@ -368,118 +365,6 @@ func (ch *clusterHandler) deployApp(appName, appTargetNamespace string, appProje
 	return appAnswers, nil
 }
 
-func (ch *clusterHandler) detectAppComponentsWhileInstall(appName, appTargetNamespace string, cluster *mgmtv3.Cluster, appAnswers map[string]string) error {
-	if cluster.Status.MonitoringStatus == nil {
-		cluster.Status.MonitoringStatus = &mgmtv3.MonitoringStatus{
-			Conditions: []mgmtv3.MonitoringCondition{
-				{Type: mgmtv3.ClusterConditionType(ConditionGrafanaDeployed), Status: k8scorev1.ConditionFalse},
-				{Type: mgmtv3.ClusterConditionType(ConditionNodeExporterDeployed), Status: k8scorev1.ConditionFalse},
-				{Type: mgmtv3.ClusterConditionType(ConditionKubeStateExporterDeployed), Status: k8scorev1.ConditionFalse},
-				{Type: mgmtv3.ClusterConditionType(ConditionPrometheusDeployed), Status: k8scorev1.ConditionFalse},
-				{Type: mgmtv3.ClusterConditionType(ConditionMetricExpressionDeployed), Status: k8scorev1.ConditionFalse},
-			},
-		}
-	}
-	monitoringStatus := cluster.Status.MonitoringStatus
-
-	enabledExporterNode := convert.ToBool(appAnswers["exporter-node.enabled"])
-	if enabledExporterNode {
-		ConditionNodeExporterDeployed.Add(monitoringStatus)
-	} else {
-		ConditionNodeExporterDeployed.Del(monitoringStatus)
-	}
-
-	enabledExporterKubeState := convert.ToBool(appAnswers["exporter-kube-state.enabled"])
-	if enabledExporterKubeState {
-		ConditionKubeStateExporterDeployed.Add(monitoringStatus)
-	} else {
-		ConditionKubeStateExporterDeployed.Del(monitoringStatus)
-	}
-
-	checkers := make([]func() error, 0, len(monitoringStatus.Conditions))
-	if !ConditionGrafanaDeployed.IsTrue(monitoringStatus) {
-		checkers = append(checkers, func() error {
-			return isGrafanaDeployed(ch.app.agentDeploymentClient, appTargetNamespace, appName, monitoringStatus, cluster.Name)
-		})
-	}
-	if enabledExporterNode && !ConditionNodeExporterDeployed.IsTrue(monitoringStatus) {
-		checkers = append(checkers, func() error {
-			return isNodeExporterDeployed(ch.app.agentDaemonSetClient, appTargetNamespace, appName, monitoringStatus)
```

Comment:
I think here is the only place we use `isNodeExporterDeployed` function. Maybe we don't need it anymore.

Reply:
Same as `isNodeExporterWithdrew`, `isGrafanaWithdrew`, `isKubeStateExporterWithdrew`, `isMetricExpressionDeployed`, `isKubeStateExporterDeployed` and `isGrafanaDeployed `.

Reply:
Updated